scripts/pre_commit_linter.py

```diff
@@ -1348,6 +1348,43 @@ def _validate_and_parse_js_files(all_files):
     return parsed_js_files
 
 
+def _get_expression(parsed_node, components_to_check):
```

Comment:
This function needs to stand on its own. Some comments:

- Please make the function name clearer: _get_expression_from_node, I think? But sometimes there is no expression so _get_expression_from_node_if_one_exists?
- Please add more detail on what the "expression part of a node" means. Pretend you're showing someone the code for the first time, and that they have no idea what it was being used for before; include all that explanation in the docstring. Your docstring can have multiple paragraphs in it.
- Be more clear about what "checking a component" means, in describing components_to_check. What is a component and what does "checking" it entail?
- A lot of the stuff in this function returns None. The docstring should explain under what circumstances this occurs, again to someone who is looking at the code for the first time. Also, why return None rather than raise Exception?

Reply:
Added more comments.

> Also, why return None rather than raise Exception?

Why should we raise Exception -- like the expression can be an identifier or just a variable initialisation expression and those are expected but we are only looking for those which are angular components and that's why returning None is better than raising an exception IMO. What do you think?

Reply:
Ah sorry I might not have been clear. I think that this is more of a question arising because I don't really understand what is going on, and the ask is for the docstring to explain that (i.e. why is returning None OK).

Again, bear in mind that the new function should stand on its own. It should not have any knowledge of what is calling it, and the docstring should be written from that perspective. The function is doing a calculation that takes in certain inputs, and must produce a certain output in response; explain clearly what the requirements for that calculation are.